pkg/server/server.go

```diff
@@ -282,8 +282,11 @@ func writeKubeConfig(certs string, config *Config) error {
 	}
 
 	if err = clientaccess.WriteClientKubeConfig(kubeConfig, url, config.ControlConfig.Runtime.ServerCA, config.ControlConfig.Runtime.ClientAdminCert,
-		config.ControlConfig.Runtime.ClientAdminKey); err != nil {
+		config.ControlConfig.Runtime.ClientAdminKey); err == nil {
+		logrus.Infof("Wrote kubeconfig %s", kubeConfig)
+	} else {
 		logrus.Errorf("Failed to generate kubeconfig: %v", err)
+		return err
```

Comment:
By propagating this back, if we fail to write the kubeconfig then k3s will exit with an error. This is new behavior, is that what we want?

Reply:
The issues I referenced all resulted in running but nonfunctional or inaccessible clusters.The current user experience is something like this:
```
ERRO[2019-03-05T05:18:01.571071590-05:00] Failed to generate kubeconfig: server https://localhost:6443/cacerts is not trusted: Get https://localhost:6443/cacerts: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "k3s-ca")
INFO[2019-03-05T05:18:01.571098190-05:00] Wrote kubeconfig /etc/rancher/k3s/k3s.yaml
INFO[2019-03-05T05:18:01.571103852-05:00] Run: k3s kubectl
INFO[2019-03-05T05:18:01.571107811-05:00] k3s is up and running
```

The success message is a straight up lie - the message is printed despite the file not being written.